### ISSUE TITLE
Fenstererkennung funktioniert nicht immer

### ISSUE THREAD

**Issue by m1chel87** [rg-engineering/ioBroker.heatingcontrol#305]
Hallo Zusammen,

ich stelle immer mal wieder fest, dass der Adapter die Fenster nicht sauber erkennt. Ich benutze ZigBee mit aqara Fensterkontakten und Eurotronic Spirit ZigBee Heizkörperventile. Im ZigBee Adapter werden die states richtig angezeigt und der heating control Adapter erkennt auch nach dem Neustart sofort, dass die Fenster offen/geschlossen sind. An was kann das liegen?

Gruß Micha

**Comment by rg-engineering** [comment]
Wir hatten hier schon das issue #265 mit ähnlichem Inhalt.
Leider kann ich das nicht nachvollziehen, benötige also mehr Infos.. Hilfreich wäre ein debug log von dem Vorgang (also wenn der Fehler auftritt)

**Comment by m1chel87** [comment]
@rg-engineering ich werde mal paar Daten sammeln. Wenn der Adapter neu gestartet ist, geht es in der Regel eine Weile. Ich kann jetzt auch nicht sagen, dass nur ein Raum betroffen ist. Ich werde es aber beobachten und mich wieder melden. Adapter 2.8.5 ist installiert

**Comment by m1chel87** [comment]
Kleines Update von mir... Ich bin noch im Daten sammeln. Ich konnte den Fehler bisher nicht nachstellen.

**Comment by rg-engineering** [comment]
@m1chel87 wäre gut zu wissen, welcher Raum, welches Fenster zu welcher Zeit (ungefähr) das Problem macht...

**Comment by m1chel87** [comment]
@rg-engineering am 24.3. ab ca. 12 Uhr. Betroffen ist das Wohnzimmer. Hatte den Fehler aber auch schon im Bad/Toilette

**Comment by rg-engineering** [comment]
Wieviele Fenster hast du für das Wohnzimmmer konfiguriert? 3? und ein SensorOpenDelay von 1200 Sek ?

Vielleicht sehe ich ein Problem...

Beispiel 13:25 Uhr:
```
2022-03-24 13:25:36.498 -  handle sensor 0 zigbee.0.00158d00041d0cf4.opened, state auto, isOpen false, new state true, openDelay 01200, closeDelay 0
2022-03-24 13:25:36.499 -  sensor open delay 1200000 for Wohnzimmer
...
2022-03-24 13:27:39.373 -  handle sensor 0 zigbee.0.00158d0007df70ef.opened, state auto, isOpen false, new state false, openDelay 01200, closeDelay 0
2022-03-24 13:27:39.373 -  cancel Open TimerId
```
Um 13:25:36 geht ein Fenster auf. Das startet den SensorOpenDelay-Timer über 1200 Sek. (=20 Minuten?) Um 13:27:39 kommt anscheinend ein Statusupdate von einem anderen Sensor und bricht den Timer ab...  

Dieses cancel  der Timer sehe ich regelmässig, damit dürfte bei dir aber die Absenkung nie (zumindest im Wohnzimmer mit mehreren Fenstern) funktionieren.... Ich sehe auch den ganzen Tag das entsprechende log nicht...

**Comment by m1chel87** [comment]
Richtig! Im Wohnzimmer sind 3 Fenster definiert. IdR wird aber nur das Rechte zum Lüften geöffnet. Das sollte eigentlich kein Problem sein, oder?

Also wenn alles passt, funktioniert die Absenkung.

**Comment by rg-engineering** [comment]
und die 20 Minuten SensorOpenDelay brauchst du?

**Comment by m1chel87** [comment]
Naja manchmal wird eben nur Mal kurz ne viertel Stunde oder so gelüftet. Da muss eben nicht wirklich die Heizung ausgeschaltet werden. Natürlich kann man das weiter reduzieren. Aber muss das nicht auch so funktionieren?

**Comment by rg-engineering** [comment]
ja, es sollte auch so funktionieren. ich versuche ja nur herauszufinden, wo das Problem ist und wie man es aktuell umgehen könnte...

**Comment by rg-engineering** [comment]
Ich denke, ich muss hier etwas umbauen... Bei Räumen mit mehreren Fenstern und einen zyklischen update des Sensorstatus und einem großen SensorDelay (größer als die zyklische Updatezeit der Sensoren) kommt es zu dem Problem.

**Comment by m1chel87** [comment]
@rg-engineering  Falls ich noch was für dich testen soll/kann, lass es mich wissen. Ich trage gerne zur Verbesserung bei

**Comment by rg-engineering** [comment]
sollte mit 2.8.6 behoben sein...

**Comment by m1chel87** [comment]
ich konnte den Fehler nicht mehr nachstellen...
